src/reducers/Material.js

```diff
@@ -42,7 +42,12 @@ function materialsToggleIsNonPeriodicForOne(state, action) {
         NPMsAlert.warning("Prevented Toggle 'isNonPeriodic' action. Please start from a cloned material");
         return state;
     }
-    newMaterial.isNonPeriodic = !newMaterial.isNonPeriodic;
+    if (newMaterial.isNonPeriodic) {
+        NPMsAlert.warning("Prevented Toggle 'isNonPeriodic' action. Structure is already 'non-periodic'");
+        return state;
+    }
+    newMaterial.isNonPeriodic = true
+    newMaterial.toCartesian();
```

Comment:
non-periodic structures coordinates make more sense in cartesian units, this also prevents the lattice size from scaling with the atom distances.